internal/testing/fixtures/generate.sh

```
@@ -0,0 +1,734 @@
+      *)
+        shift
+        ;;
+    esac
+  done
+
+  mkdir -p -- "$(dirname "$output_file")"
+  
+  # Create new JSON or merge with existing file
+  if [[ -f "$output_file" ]]; then
+    # First create the new JSON structure
+    local new_json=$(jq -n "${jq_args[@]}" "$jq_query")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)